src/main.c

```diff
@@ -184,47 +177,27 @@ void process_input(numberstack* numbers, operation** current_op, char* in) {
          * 3 - a symbol then a number i.e. "+2"
          * 4 - a number, a symbol and a number i.e. "2+2"
          *
-         * The following code handles the input to generate one of those cases
-         * Adding the numbers and operation accordingly to the respective structures
+         * The following code, first, handles a number that comes before the operation
+         * And then, in the while loop, handles the rest of the input that starts with an operation
          */
 
-        // Find the first number in the string, by splitting with the symbol found
-        char * token = strtok(in, opchar);
+        /* Before the strtok replaces the operator with \0 when searching for tokens
+         * Save the string starting at the operator so we can reuse after handling the first number
+         */
+        char * in_saved = strdup(op);
```

Comment:
in_saved is never `free()`

Reply:
I've fixed this @DevManu-de. Do you think you could check it one more time?
I believe it's ready to merge.
Afterwards you could retry the memory leak tests with the merged version as well.